### DISCUSSION TITLE
Performance regression

### DISCUSSION THREAD

**Discussion by thibthib** [vercel/turborepo#5604] [Help] [answered]
Hello 👋

I upgraded our turbo version from `1.6.1` to `1.10.11` with `npx @turbo/codemod migrate` and we're now seeing a quite slower execution time:

Before:
<img width="320" alt="image" src="https://github.com/vercel/turbo/assets/1030644/bcdcb43c-e3fb-4a8c-b901-8539ec2b6096">

After:
<img width="338" alt="image" src="https://github.com/vercel/turbo/assets/1030644/9ce60ea8-d113-4c3f-98b9-24d0a531c889">

This is also visible in our monitoring tools:
![Turbo duration](https://github.com/vercel/turbo/assets/1030644/2a6987e3-bdc2-4386-988b-4a0b8fbfa5bd)

Should I open an issue ? Or is this possibly something on our end, with some turbo configuration change needed after the upgrade ?

**Comment by anthonyshew** [comment]
This is likely a result of the porting effort from Go to Rust that we finished in 1.11 last week. You're likely seeing the effects of having to pass data between the two binaries - but 1.11 is all Rust now!

Could you try upgrading your repo to latest? If you're still seeing regressions, please open an issue. 🙏

**Reply by thibthib** [reply]
We suspected the data transfer would be the culprit: with almost a thousand packages now, the time to display `Packages in scope` was taking more than 15s. Sadly, we upgraded to 1.11.1 on Friday and did not see any performance improvement though: 

<img width="330" alt="image" src="https://github.com/vercel/turbo/assets/1030644/7db20246-a5af-457a-bd5a-32f1acf053d8">

The thing which is weird now: the `Packages in scope` message now takes only a few seconds to display, much faster than before ! So I assume there's a regression in the cache fetching mechanism or something in the package loop that makes it slower.

**Reply by thibthib** [reply]
Another data point: locally with a hot cache and remote cache disabled, it seems indeed faster than before:
<img width="270" alt="image" src="https://github.com/vercel/turbo/assets/1030644/bceb5a42-b534-42bb-a3cc-43209af9d336">

**Reply by anthonyshew** [reply]
This is really great info. Thank you.

We're _incredibly, incredibly_ interested in a trace if you'd be able to get us one. You can produce one using:
```
turbo <your tasks> --anon-profile=proflie.json
```
 
Would you be able to send it to anthony.shew@vercel.com? 🙏

**Comment by anthonyshew** [comment, marked as answer]
Thank you for closing! This regression is being investigated.

If anyone else out there is stumbling across performance regressions, we would appreciate it if you sent an email to anthony.shew@vercel.com including:

- The profile.json once you've ran `turbo run <your tasks> --anon-profile=profile.json`
- The environment you're running the command in
- A short description of the tasks you ran (building applications, running ESLint, running codegen tasks)
